From 589f06bc02205d6ec3c8b21c463174b3bd81ca4a Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Mon, 30 Jul 2018 14:50:24 +0800
Subject: [PATCH 57/65] do not customize window theme

The customized window theme may broke metacity.
...
|(metacity:2675): metacity-WARNING **: "(null)" found in configuration
database is not a valid value for mouse button modifier
...

Also remove install-data-hook, which caused host contamination

Upstream-Status: Inappropriate [oe specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>

Rebase for anaconda 34.

Signed-off-by: Kai Kang <kai.kang@windriver.com>

Rebase for anaconda 37. And it replaces metacity with gnome-kiosk in
upstream. Revert it and turn back to metacity.

Signed-off-by: Kai Kang <kai.kang@windriver.com>

Rebase for anaconda 38 on 20231107.

Signed-off-by: Kai Kang <kai.kang@windriver.com>
---
 data/window-manager/config/Makefile.am | 6 +-----
 pyanaconda/display.py                  | 8 ++++----
 2 files changed, 5 insertions(+), 9 deletions(-)

diff --git a/data/window-manager/config/Makefile.am b/data/window-manager/config/Makefile.am
index 90e71d9161..5d7a2b80e1 100644
--- a/data/window-manager/config/Makefile.am
+++ b/data/window-manager/config/Makefile.am
@@ -26,11 +26,7 @@ dist_schema_DATA = org.gnome.desktop.wm.keybindings.gschema.override \
 # schemas they modify, so pretend that this is the case with symlinks and
 # create the compiled schema.
 install-data-hook:
-	$(MKDIR_P) $(DESTDIR)$(schemadir)
-	$(LN_S) -f /usr/share/glib-2.0/schemas/org.gnome.desktop.wm.keybindings.gschema.xml $(DESTDIR)$(schemadir)
-	$(LN_S) -f /usr/share/glib-2.0/schemas/org.gnome.desktop.wm.preferences.gschema.xml $(DESTDIR)$(schemadir)
-	$(LN_S) -f /usr/share/glib-2.0/schemas/org.gnome.desktop.enums.xml $(DESTDIR)$(schemadir)
-	glib-compile-schemas --strict $(DESTDIR)$(schemadir)
+	echo "pass"
 
 uninstall-local:
 	rm -f $(DESTDIR)$(schemadir)/*.xml
diff --git a/pyanaconda/display.py b/pyanaconda/display.py
index 5e14af16bc..d443500a3f 100644
--- a/pyanaconda/display.py
+++ b/pyanaconda/display.py
@@ -204,16 +204,16 @@ def do_startup_x11_actions():
     if 'XDG_DATA_DIRS' in os.environ:
         xdg_data_dirs = datadir + '/window-manager:' + os.environ['XDG_DATA_DIRS']
     else:
-        xdg_data_dirs = datadir + '/window-manager:/usr/share'
+        xdg_data_dirs = '/usr/share'
 
     def x11_preexec():
         # to set GUI subprocess SIGINT handler
         signal.signal(signal.SIGINT, signal.SIG_IGN)
-        
-    childproc = util.startProgram(["gnome-kiosk", "--display", ":1", "--sm-disable", "--x11"],
+
+    childproc = util.startProgram(["metacity", "--display", ":1", "--sm-disable"],
                                   env_add={'XDG_DATA_DIRS': xdg_data_dirs},
                                   preexec_fn=x11_preexec)
-    WatchProcesses.watch_process(childproc, "gnome-kiosk")
+    WatchProcesses.watch_process(childproc, "metacity")
 
 
 def set_x_resolution(runres):
-- 
2.7.4
